cmd/gate/root.go

```diff
@@ -47,22 +48,35 @@ func Execute() {
 
 func init() {
 	cobra.OnInitialize(initConfig)
+	v := gate.Viper
 
+	rootCmd.PersistentFlags().String("config", "", "config file (default is ./config.yaml)")
 	rootCmd.PersistentFlags().BoolP("debug", "d", false, "Enable debug mode")
+	v.BindPFlag("config", rootCmd.PersistentFlags().Lookup("config"))
+	v.BindPFlag("debug", rootCmd.PersistentFlags().Lookup("debug"))
 }
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
 	v := gate.Viper
-	_ = v.BindPFlag("debug", rootCmd.Flags().Lookup("debug"))
 
+	// Load Environment Variables
 	v.SetEnvPrefix("GATE")
 	v.AutomaticEnv() // read in environment variables that match
 	v.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
 
-	v.SetConfigFile("config.yml")
+	if cfgFile := v.GetString("config"); cfgFile != "" {
+		v.SetConfigFile(cfgFile)
+	} else {
+		v.SetConfigName("config")
+		v.AddConfigPath(".")
+	}
+
 	// If a config file is found, read it in.
-	if err := v.ReadInConfig(); err == nil {
+	err := v.ReadInConfig()
+	if err != nil {
+		fmt.Println(fmt.Errorf("Error reading config file: %s \n", err))
```

Comment:
Gate supports configuration by env variables and does not require a config file.
Could you please ignore the error and maybe add a comment that a config file is not required?

Reply:
I realized that at one point... but then I forgot _doh!_

Reply:
Counterpoint:  What about situations where the error is due to malformed configuration files?  That seems useful, I think only the "not found" should be ignored.